debian/gen-flat-repo-fixtures.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+set -eux
+
+yes | apt-get install dpkg-dev dpkg-sig apt-utils
+
+SRCDIR=$(readlink -f "$(dirname "$0")")
+OUTPUTDIR=$(realpath "${SRCDIR}/../$1")
+
+mkdir "${OUTPUTDIR}"
+
+cd "$OUTPUTDIR"
+(
+  for CTL in "${SRCDIR}"/asgard/*.ctl
+  do
+    equivs-build --arch ppc64 "${CTL}"
+  done
+)
+
+curl -L https://github.com/pulp/pulp-fixtures/raw/master/common/GPG-PRIVATE-KEY-pulp-qe | gpg --import
+echo "6EDF301256480B9B801EBA3D05A5E6DA269D9D98:6:" | gpg --import-ownertrust
+
+dpkg-sig --sign builder ./*.deb
+
+apt-ftparchive packages ../debian-flat > Packages
+sed -i 's!../debian-flat!debian-flat!' ./Packages
```

Comment:
Why is this manipulation needed? Can't you just `cd ../debian-flat` and call `apt-ftparchive` there?

BTW: I thought a flat repo was created with `dpkg-scanpackages | gz > Packages.gz` but that might be outdated information.

Reply:
The problem is, that the Packages file still has the values for path starting with "./" which I need to replace with "debian-flat/", otherwise pulp cannot find the packages. One could also go inside the folder, create the Packages file, change the entries and go outside again...
I think one could also use dpkg-scanpackages, but as I understand the manpage, that only creates a .gz or .xz file. Please see https://pulp.plan.io/issues/7768?issue_count=26&issue_position=1&next_issue_id=7756 as a related issue. I had troubles synchronising repositories in pulp_deb which have only a Packages.gz or Packages.xz file.

Reply:
What i am thinking here is, "If the flat-repo-format is a _thing_, there must be tools around to create it. And those tools would not require you to make such strange manipulations". So i urge you to find those tools (no, i don't mean pulp XD).

On the other hand changing this fixture in a way that pulp can sync it is really the wrong direction.

The fixture should be reflecting a format that is valid and consumable by apt. And once you have that you try to make pulp understand it. And as far as i remember, the easiest way to create a repo is:
Throw some packages in a directory and call `dpkg-scanpackages` (It might depend on where you call it how it references the files relatively.)

Maybe the one including a `Release` file is already a different format.

https://linux.die.net/man/1/dpkg-scanpackages

Reply:
Thanks for your ideas! Still, there are some issues for me that I cannot quite see through:

1) I don't really think that flat repo format is "the big thing". At least, it's not recommended as "best practice". Still, some people use it to build packages manually, at least I could not find any app that creates packages in flat repo format. If someone knows how to use an app, they would use an app that creates a nice repo ie not flat, wouldn't they? Still, I changed the generation as I would do it manually. What do you think? Not really satisfying, is it?
Why I need to change the path for pulp, I don't really get either. For the bash commands, I could insert as many "./" as I want in a path and it still can find the correct path. However, Pulp cannot deal with "./" in a path, or rather the os.path python module then maybe returns wrong file paths or components... But maybe I can change that in the synchronizing.py somehow...

2) Do the SHA and MD5 values depend on which folder you are in when building the Packages file? When I tried it (once from outside, once from inside the folder), the SHA and MD5 did not change...

2) The flat repos I've found in the Internet so far all have Release files, so I think we can expect that. Also the debian repo standard requires a Release file. What do you think?

Thanks for your help!